### PR TITLE
Separate out logic from main.rs to lib.rs

### PR DESCRIPTION
**Pull request by 2ameya**
Other misc things:
a. Rename hello-world to hello_world
b. Implement basic error handling

### CODE REVIEW

File: hello_world/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "hello-world"
+name = "hello_world"
```
Comment (anishkny) [inline hello_world/Cargo.toml:2]: Good. Follows [naming conventions](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html).

File: hello_world/Cargo.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
 version = "0.1.0"
 authors = ["Ameya Palande <2ameya@gmail.com>"]
```
Comment (anishkny) [inline hello_world/Cargo.toml:4]: Feel free to add `"Anish Karandikar <anishkny@gmail.com>"` 😄

**Review by anishkny — APPROVED** [review]
What do you feel about 2 spaces indentation? Allows you to fit more stuff horizontally on screen. And can be enforced via:

```toml
# .rustfmt.toml
tab_spaces=2
```

**Comment by 2ameya** [comment]
> What do you feel about 2 spaces indentation? Allows you to fit more stuff horizontally on screen. And can be enforced via:
> 
> ```toml
> # .rustfmt.toml
> tab_spaces=2
> ```

I guess your next PR will do this change and then let's reformat the whole source with the new setting. I have no problem with any setting for number of spaces.

File: hello_world/Cargo.toml (hunk 2 above)
Comment (2ameya) [reply hello_world/Cargo.toml:4]: Feel free to add it in your next PR.

File: hello_world/Cargo.toml (hunk 1 above)
Comment (2ameya) [reply hello_world/Cargo.toml:2]: Wasn't aware of the naming convention. While compiling realized that rustc is automatically converting "-" to "_".

**Pull request merged**